docs/pages/hooks/useAddressInput.mdx

```diff
@@ -0,0 +1,194 @@
+---
+layout: docs
+---
+
+import React from "react";
+import { DocsProvider } from "../../components/DocsProvider";
+import {
+  UseAddressInputBasicExample,
+  UseAddressInputWithResolutionExample,
+  UseAddressInputErrorExample
+} from "../../components/UseAddressInputExamples";
+
+# useAddressInput Hook
+
+The `useAddressInput` hook handles address input with debouncing, ENS name resolution (both directions), and ENS avatar fetching. It's designed for building custom address input components with real-time ENS lookups.
+
+## Import
+
+```tsx
+import { useAddressInput } from "@scaffold-ui/hooks";
+```
+
+## Usage
+
+```tsx
+const {
+  ensAddress,
+  ensName,
+  ensAvatar,
+  isEnsAddressLoading,
+  isEnsNameLoading,
+  isEnsAvatarLoading,
+  settledValue,
+  debouncedValue
+} = useAddressInput({
+  value: inputValue,
+  debounceDelay: 500
+});
+```
+
+## Parameters
+
+| Parameter        | Type                   | Default | Description                                              |
+| ---------------- | ---------------------- | ------- | -------------------------------------------------------- |
+| `value`          | `string`               | -       | The input value (can be an Ethereum address or ENS name).|
+| `debounceDelay`  | `number` *(optional)*  | `500`   | Debounce delay in milliseconds for ENS resolution.       |
+
+## Return Values
+
+| Property                | Type                  | Description                                                        |
+| ----------------------- | --------------------- | ------------------------------------------------------------------ |
+| `ensAddress`            | `Address \| undefined` | The resolved Ethereum address from an ENS name input.              |
```

Comment:
type is `GetEnsAddressReturnType | undefined` where `GetEnsAvatarReturnType` is `Address | null`, so result is `Address | null | undefined`